packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_stats_table.dart

```diff
@@ -0,0 +1,169 @@
+      : super(
+          'Instances',
+          titleTooltip: 'Number of instances of the class\n'
+              'retained by the path.',
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(_RetainingPathRecord record) => record.value.instanceCount;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+}
+
+class _ShallowSizeColumn extends ColumnData<_RetainingPathRecord> {
+  _ShallowSizeColumn()
+      : super(
+          'Shallow\nDart Size',
+          titleTooltip: shallowSizeColumnTooltip,
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(_RetainingPathRecord record) => record.value.shallowSize;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+
+  @override
+  String getDisplayValue(_RetainingPathRecord record) => prettyPrintBytes(
+        getValue(record),
+        includeUnit: true,
+        kbFractionDigits: 1,
+      )!;
+}
+
+class _RetainedSizeColumn extends ColumnData<_RetainingPathRecord> {
+  _RetainedSizeColumn()
+      : super(
+          'Retained\nDart Size',
+          titleTooltip: retainedSizeColumnTooltip,
+          fixedWidthPx: scaleByFontFactor(85.0),
+          alignment: ColumnAlignment.right,
+        );
+
+  @override
+  int getValue(_RetainingPathRecord record) => record.value.retainedSize;
+
+  @override
+  bool get supportsSorting => true;
+
+  @override
+  bool get numeric => true;
+
+  @override
+  String getDisplayValue(_RetainingPathRecord record) => prettyPrintBytes(
+        getValue(record),
+        includeUnit: true,
+        kbFractionDigits: 1,
+      )!;
+}
+
+class ClassStatsTable extends StatefulWidget {
+  const ClassStatsTable({
+    Key? key,
+    required this.data,
+    required this.sorting,
+  }) : super(key: key);
+
+  final HeapClassStatistics data;
+  final ColumnSorting sorting;
+
+  @override
+  State<ClassStatsTable> createState() => _ClassStatsTableState();
+}
+
+class _ClassStatsTableState extends State<ClassStatsTable>
+    with AutoDisposeMixin {
+  late final List<ColumnData<_RetainingPathRecord>> _columns;
+
+  @override
+  void initState() {
+    super.initState();
+
+    final _shallowSizeColumn = _ShallowSizeColumn();
+
+    _columns = <ColumnData<_RetainingPathRecord>>[
+      _RetainingPathColumn(),
+      _InstanceColumn(),
+      _shallowSizeColumn,
+      _RetainedSizeColumn(),
+    ];
+
+    if (!widget.sorting.initialized) {
+      widget.sorting
+        ..direction = SortDirection.descending
+        ..columnIndex = _columns.indexOf(_shallowSizeColumn)
+        ..initialized = true;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatTable<_RetainingPathRecord>(
+      columns: _columns,
+      data: widget.data.byRetainingPath.entries.toList(growable: false),
```

Comment:
do this in initState (and didUpdateWidget) so we don't have to copy a list on every build.